packages/nx-plugin/src/generators/add-to-project/generator.unit.test.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Tree,
+  addProjectConfiguration,
+  readProjectConfiguration,
+} from '@nx/devkit';
+import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
+import { join } from 'path';
+import { describe, expect, it } from 'vitest';
+import { addToProjectGenerator } from './generator';
+
+describe('init generator', () => {
+  let tree: Tree;
+  const testProjectName = 'test-app';
+
+  beforeEach(() => {
+    tree = createTreeWithEmptyWorkspace();
+
+    addProjectConfiguration(tree, testProjectName, {
+      root: `apps/${testProjectName}`,
+      projectType: 'library',
+      sourceRoot: `apps/${testProjectName}/src`,
```

Comment:
We should check for the workspace layout. In our repo e.g. we have no apps/libs but packages/examples/e2e

Reply:
This may need to be clarified. We just read the project's configuration and added the code-pushup configuration to its root. If the project root is the same as the repo root, we add the `code-pushup` config to the repo root. Any other logic here may be extra.